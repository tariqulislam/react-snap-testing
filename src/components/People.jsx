import React from 'react'
import PropTypes from 'prop-types'

function People (props) {
    const {item} = props
    return (
       <tr>
           <td>{item.id}</td>
           <td>{item.username}</td>
           <td>{item.email}</td>
           <td>{item.phone}</td>
       </tr>
    )
}

People.propTypes = {
    item: PropTypes.object,
}

export default People;