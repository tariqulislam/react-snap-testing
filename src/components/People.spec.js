import React from 'react'
import renderer from 'react-test-renderer'
import People from './People'

describe(`Test People Information in row`, () => {
  it('renders the row without people inforamtion', () => {
      const tree = renderer.create(<People />).toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('renders the row with people information', () => {
      const people = { id: 1, username: 'tariqulislam', email: 'tariqul.islam.rony@gmail.com', phone: 'phone'};
      const tree = renderer.create(<People item={people}  />).toJSON();
      expect(tree).toMatchSnapshot();
  })
});