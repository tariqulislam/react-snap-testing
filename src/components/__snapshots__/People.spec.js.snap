// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test People Information in row renders the row with people information 1`] = `
<tr>
  <td>
    1
  </td>
  <td>
    tariqulislam
  </td>
  <td>
    tariqul.islam.rony@gmail.com
  </td>
  <td>
    phone
  </td>
</tr>
`;

exports[`Test People Information in row renders the row without people inforamtion 1`] = `
<span>
  No Item Found
</span>
`;
